newrelic/common/agent_http.py

```diff
@@ -524,24 +524,31 @@ def _supportability_request(params, payload, body, compression_time):
         # *********
         # Used only for supportability metrics. Do not use to drive business
         # logic!
+        # payload: uncompressed
+        # body: compressed
         agent_method = params and params.get("method")
         # *********
 
-        if agent_method and body:
+        if agent_method and payload:
             # Compression was applied
             if compression_time is not None:
                 internal_metric(
-                    "Supportability/Python/Collector/ZLIB/Bytes/%s" % agent_method,
-                    len(payload),
+                    "Supportability/Python/Collector/%s/ZLIB/Bytes" % agent_method,
```

Comment:
It might be better to add this in a commit message but for my own sake and the sake of any future developer reading this, I'm going to explain what's going on here as it wasn't immediately clear just reading the code:

We previously were not following the agent spec in a couple of ways which this code now fixes:
1. The method name should come before the /ZLIB/Bytes path: `Supportability/Python/Collector/ZLIB/Bytes/%s` -> `Supportability/Python/Collector/%s/ZLIB/Bytes`
1. The spec wants a count independent of the method: `Supportability/Python/Collector/ZLIB/Bytes`.